internal/vm-console-proxy-bundle/bundle.go

```diff
@@ -27,117 +28,100 @@ var (
 	return filepath.Join(vmConsoleProxyBundleDir, "vm-console-proxy.yaml")
 }
 
-func readFile(fileName string) ([][]byte, error) {
-	file, err := os.ReadFile(fileName)
-	if err != nil {
-		return nil, err
-	}
+func loadBundleFromBytes(data []byte) (*Bundle, error) {
+	bundle := &Bundle{}
+	decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(data), 1024)
+	for {
+		obj := &unstructured.Unstructured{}
+		err := decoder.Decode(&obj)
+		if err == io.EOF {
+			return bundle, nil
+		}
+		if err != nil {
+			return nil, err
+		}
 
-	return [][]byte{file}, nil
-}
+		kind := obj.GetKind()
+		if kind == "" {
+			continue
+		}
 
-func decodeObjectsFromFiles(files [][]byte) (*Bundle, error) {
-	bundle := &Bundle{}
-	for _, file := range files {
-		decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(file), 1024)
-		for {
-			var obj map[string]interface{}
-			err := decoder.Decode(&obj)
-			if err == io.EOF {
-				break
-			}
-			if err != nil {
-				return nil, err
-			}
-			if kind, ok := obj["kind"].(string); ok {
-				if kind == "" {
-					continue
-				}
+		var destObj any
+		switch kind {
+		case rbac.ServiceAccountKind:
+			destObj = &(bundle.ServiceAccount)
+		case clusterRoleKind:
+			destObj = &(bundle.ClusterRole)
+		case clusterRoleBindingsKind:
+			destObj = &(bundle.ClusterRoleBinding)
+		case serviceKind:
+			destObj = &(bundle.Service)
+		case deploymentKind:
+			destObj = &(bundle.Deployment)
+		case configMapKind:
+			destObj = &(bundle.ConfigMap)
+		default:
+			return nil, fmt.Errorf("unsupported Kind found in vm-console-proxy bundle: %s", kind)
+		}
 
-				switch kind {
-				case serviceAccountKind:
-					serviceAccount := core.ServiceAccount{}
-					err = getObject(obj, &serviceAccount)
-					if err != nil {
-						return nil, err
-					}
-					bundle.ServiceAccount = serviceAccount
-				case clusterRoleKind:
-					clusterRole := rbac.ClusterRole{}
-					err = getObject(obj, &clusterRole)
-					if err != nil {
-						return nil, err
-					}
-					bundle.ClusterRole = clusterRole
-				case clusterRoleBindingsKind:
-					clusterRoleBinding := rbac.ClusterRoleBinding{}
-					err = getObject(obj, &clusterRoleBinding)
-					if err != nil {
-						return nil, err
-					}
-					bundle.ClusterRoleBinding = clusterRoleBinding
-				case serviceKind:
-					service := core.Service{}
-					err = getObject(obj, &service)
-					if err != nil {
-						return nil, err
-					}
-					bundle.Service = service
-				case deploymentKind:
-					deployment := apps.Deployment{}
-					err = getObject(obj, &deployment)
-					if err != nil {
-						return nil, err
-					}
-					bundle.Deployment = deployment
-				case configMapKind:
-					configMap := core.ConfigMap{}
-					err = getObject(obj, &configMap)
-					if err != nil {
-						return nil, err
-					}
-					bundle.ConfigMap = configMap
-				default:
-					continue
-				}
-			}
+		err = runtime.DefaultUnstructuredConverter.FromUnstructured(obj.UnstructuredContent(), destObj)
+		if err != nil {
```

Comment:
What about make it shorter:
```
if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.UnstructuredContent(), destObj); err != nil {
	return nil, err
}
```

Reply:
The line is already quite long. I prefer it this way.